app/models/authority.rb

```diff
@@ -334,11 +338,8 @@ def publish_if_first!
     publish! if awaiting_first?
   end
 
-  def root_collection
-    return @root_collection if @root_collection
-    return @root_collection = Collection.find(root_collection_id) if root_collection_id
-
-    return @root_collection = generate_root_collection!
+  def obtain_root_collection
```

Comment:
root_collection method was very misleading, as it hid actual database relation. So I've added belongs_to root_collection, and renamed this method to not hide relation.